libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/ConnectivityStatusProvider.java

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+class ConnectivityStatusProvider {
+
+  private final Context context;
+  private final ConnectivityStatusMap statusMap;
+
+  ConnectivityStatusProvider(Context context) {
+    this.context = context;
+    this.statusMap = new ConnectivityStatusMap();
+  }
+
+  boolean isConnected() {
+    NetworkInfo info = getNetworkInfo(context);
+    return (info != null && info.isConnected());
+  }
+
+  boolean isConnectedFast() {
+    NetworkInfo info = getNetworkInfo(context);
+    return (info != null && info.isConnected() && isConnectionFast(info.getType(), info.getSubtype()));
+  }
+
+  @SuppressLint("MissingPermission")
+  private NetworkInfo getNetworkInfo(Context context) {
+    ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+    return cm.getActiveNetworkInfo();
+  }
+
+  private boolean isConnectionFast(int type, int subType) {
+    if (type == ConnectivityManager.TYPE_WIFI) {
```

Comment:
We should be careful here because you could be connected to a Wi-Fi but that doesn't mean to have connectivity e.g. you could be connected to a hotspot and don't have connection until you log in.

I believe we should implement something similar as with mobile connectivity. Maybe 👀 Wi-Fi strength?

Reply:
This is good to think about as an improvement but can possibly be deferred.

Reply:
I added a wifi strength "checker" to address this